### ISSUE TITLE
weights initialization

### ISSUE THREAD

**Issue by rola93** [divamgupta/image-segmentation-keras#304]
Hello, I'm trying to understand how does the weights get initialized. 

There are several implementations for encoders (mobilenet, vgg16, resnet) which  are implemented on diferent modules, providing a similar operation for all of them, as follows:

```python
def get_resnet50_encoder(input_height=224,  input_width=224,
                         pretrained='imagenet',
                         include_top=True, weights='imagenet',
                         input_tensor=None, input_shape=None,
                         pooling=None,
                         classes=1000, channels=3):
```

However, those `pretrained` weights are only used internally, most of the time just with default values. In the snippet I included before, [taken from here](https://github.com/divamgupta/image-segmentation-keras/blob/dc830bbd76371aaedbf8cb997bdedca388c544c4/keras_segmentation/models/resnet50.py#L122), the `pretrained` parameter is only used as follows:
```python
if pretrained == 'imagenet':
    weights_path = keras.utils.get_file(
        pretrained_url.split("/")[-1], pretrained_url)
    Model(img_input, x).load_weights(weights_path, by_name=True, skip_mismatch=True)
```

my question is the following:
1) Are those weights having effect? is the model actually being initialized with those values? It is strange since `Model` object in the snippet isn't used any more.
2) When I run the following:

```python
from keras_segmentation.models.unet import resnet50_unet

model = resnet50_unet(n_classes=n_classes)
```

Are encoder's weights randomly initialized or they are set to imagenet ones? How can I use them with random initialization?

Thanks in advance

**Comment by divamgupta** [comment]
Hi,

As of now it will initialize from the imagenet weights itself. Yes, it is a good idea to have that flag at the model functions as well.

**Comment by andrey-bat** [comment]
It seems to me, those weights are not really loaded into model, at least in environments I tried (Kaggle, Colab, local win10).
Is there some way to test, if they're loaded?

**Comment by rola93** [comment]
I tried it locally in Ubuntu / jupyter notebook, and all checked were some weights values in the encoder. After several initializations they didn't change, but for some others in the decoder they do.

I know it isn't a formal proof but maybe you can try it...

**Comment by andrey-bat** [comment]
> I tried it locally in Ubuntu / jupyter notebook, and all checked were some weights values in the encoder. After several initializations they didn't change, but for some others in the decoder they do.

The fact, they did not change implies they're not random... Yes, they could be generated with same seed, but using fixed seed is unlikely, when initializing weights...
However, I'm going to check if they are distributed normally.
